rest/persistent_config_test.go

```diff
@@ -0,0 +1,122 @@
+package rest
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAutomaticConfigUpgrade(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("CBS required")
+	}
+
+	tb := base.GetTestBucket(t)
+	defer tb.Close()
+
+	rawConfig := `
+	{
+		"interface": ":4444",
+  		"adminInterface": ":4445",
+		"databases": {
+			"db": {
+			  "server": "%s",
+			  "username": "%s",
+			  "password": "%s",
+			  "bucket": "%s"
+			}
+		}
+	}`
+
+	config := fmt.Sprintf(rawConfig, base.UnitTestUrl(), base.TestClusterUsername(), base.TestClusterPassword(), tb.GetName())
+
+	tmpDir, err := ioutil.TempDir("", t.Name())
+	require.NoError(t, err)
+
+	configPath := filepath.Join(tmpDir, "config.json")
+	err = ioutil.WriteFile(configPath, []byte(config), os.FileMode(0644))
+	require.NoError(t, err)
+
+	startupConfig, _, err := automaticConfigUpgrade(configPath)
+	assert.NoError(t, err)
+
+	assert.Equal(t, base.UnitTestUrl(), startupConfig.Bootstrap.Server)
+	assert.Equal(t, base.TestClusterUsername(), startupConfig.Bootstrap.Username)
+	assert.Equal(t, base.TestClusterPassword(), startupConfig.Bootstrap.Password)
+	assert.Equal(t, ":4444", startupConfig.API.PublicInterface)
+	assert.Equal(t, ":4445", startupConfig.API.AdminInterface)
```

Comment:
Are we able to include assertions for the backup file and newly written config file too? Not just the returned in-memory versions.